src/Data/FunctorF.purs

```diff
@@ -0,0 +1,17 @@
+module Data.FunctorF where
+
+import Data.NaturalTransformation (type (~>))
+
+-- | A type class for higher order functors; types of kind `(* -> *) -> * -> *`.
+-- |
+-- | The mapping operation `hoist` allows the inner functor to be changed. This
+-- | is similar to `map` for a `Functor`, only here we are transforming a
+-- | functor rather than a value, so correspondingly we perform this with a
+-- | natural transformation rather than a rank-1 function.
+-- |
+-- | Instances must satisfy the following laws:
+-- |
+-- | - Identity: `hoist id = id`
+-- | - Composition: `hoist (f <<< g) = hoist f <<< hoist g`
+class FunctorF t where
+  hoist :: forall f g. (f ~> g) -> t f ~> t g
```

Comment:
Well, I was looking at it this way too: we have plenty of functor hoists defined, and not one mmorph hoist currently, and in fact, no representation of monad morphisms at all. I'd be fine with import `MM.hoist` and `FF.hoist` as different things if it does turn out the mmorph version is actually useful.